### PR TITLE
Remove Flow support in React preset

### LINKED ISSUE #6095: babel-preset-react includes Flow, which conflicts with babel-preset-typescript.
When trying out babel-preset-typescript with babel-preset-react, I learned that the React preset comes bundled with Flow. This hasn't caused any problems in Babel <7, as Flow doesn't conflict with anything else.

With Babel 7 and the Typescript preset, the current experience is to see an error about the conflict between Flow and Typescript, then search the documentation for each of the presets used, figure out which one is including Flow, and replace it with the three React plugins.

```js
{
  plugins: [
    "syntax-jsx",
    "transform-react-jsx",
    "transform-react-display-name"
  ],
  presets: ["typescript"],
}
```

My question is: should Flow still be bundled in this preset? Or, should the error message describe what is causing Flow to be included? It's strange to have a type system unrelated to React to be included in the React preset. This was a quick fix for me, but the "me" from one year ago may have spent hours trying to figure out why Typescript and React together result in an error.

### Babel Configuration (.babelrc, package.json, cli command)

```js
{
  presets: [
    "typescript",
    "react"
  ]
}
```

### Current Behavior
Error: "Cannot combine flow and typescript plugins".

### Possible Solution
Include extra information in error message (flow is included in babel-preset-react, use React plugins instead)
Remove preset-flow from babel-preset-react in v7.

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            | 7.0.0-alpha.19
| node             | 8.1.4
| npm              | 5.0.3
| Operating System | Windows 10

### PR DESCRIPTION
**Pull request by ramasilveyra**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues             | Fixes #6095
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  | Y
| Minor: New Feature?      | N
| Tests Added/Pass?        | Y/Y
| Spec Compliancy?         | N
| License                  | MIT
| Doc PR                   | 
| Any Dependency Changes?  | 

Remove Flow support in React preset. Needed to fix an incompatibility issue with typescript preset, another option could be adding a [`noFlow` option](https://github.com/babel/babel/issues/6095#issuecomment-322566079).

### CODE REVIEW

**Comment by xtuc** [comment]
Would be ideal but not a good idea to me because too many users are relying on this behavior.

**Comment by chicoxyzzy** [comment]
Why can't we introduce this breaking change in 7.0?

**Comment by xtuc** [comment]
Yes, we could. Let's discuss it at the weekly later today.

**Comment by hzoo** [comment]
Also via https://twitter.com/AdamRackis/status/898568493896839169 and other reports

File: packages/babel-preset-react/README.md [hunk 1]
```diff
@@ -5,7 +5,6 @@
 This preset always includes the following plugins:
 
 - [syntax-jsx](https://babeljs.io/docs/plugins/syntax-jsx/)
-- [transform-flow-strip-types](https://babeljs.io/docs/plugins/transform-flow-strip-types/)
```
Comment (existentialism) [inline packages/babel-preset-react/README.md]: Wonder if it's worth including a note about how Flow syntax support is no longer enabled in 7.x?

Comment (ramasilveyra) [inline packages/babel-preset-react/README.md]: Yes, it's worth it. Added this:
```
> Note: Flow syntax support is no longer enabled in v7. For that, you will need to add the [Flow preset](https://babeljs.io/docs/plugins/preset-flow/).
```

**Comment by ramasilveyra** [comment]
The Circle CI build started to fail on the `yarn` step with this error:
```
error An unexpected error occurred: "EEXIST: file already exists, mkdir '/home/ubuntu/babel/node_modules/babel-plugin-transform-export-extensions'".
```
And I cannot reproduce on my side, any ideas?

File: packages/babel-plugin-transform-flow-strip-types/src/index.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ export default function({ types: t }) {
         if (skipStrip) {
           throw path.buildCodeFrameError(
             "A @flow directive is required when using Flow annotations with " +
-              "babel-preset-react or the `requireDirective` option.",
+              "the `requireDirective` option.",
```
Comment (ramasilveyra) [inline packages/babel-plugin-transform-flow-strip-types/src/index.js]: Also I just found this, I think that it should be changed in this pr too. If i'm wrong please ping me and I will remove this changes.

**Comment by nicolo-ribaudo** [comment]
@ramasilveyra I restarted the Circle CI build, now it passed.

**Comment by hzoo** [comment]
@xtuc Was always for 7.0, not planning to do 6.x releases 👍

**Comment by hzoo** [comment]
testing in the repl for this build http://babeljs.io/repl/build/4665#?evaluate=false&lineWrap=true&presets=es2015%2Creact%2Cstage-0&babili=false&builtIns=false&debug=false&code_lz=PTAEAECsGcA9QIbQCYDMBQ6BuCBOoBTUAXlAB5kBLLAPjOgAcEA7YO4K2zVAV2YGMALpQD2zRAAoEALlDRBuSswDmASlABvAL5A&experimental=false&loose=false&spec=false&playground=true

I guess we need flow/typescript in the repl options now 😄

**Pull request merged**